pay-api/src/pay_api/services/internal_pay_service.py

```diff
@@ -116,3 +115,27 @@ def complete_post_invoice(self, invoice: Invoice, invoice_reference: InvoiceRefe
             }
         )
         transaction.update_transaction(transaction.id, pay_response_url=None)
+
+    @staticmethod
+    def _validate_routing_slip(routing_slip: RoutingSlipModel, invoice: Invoice):
+        """Validate different conditions of a routing slip payment."""
+        # is rs doesnt exist , legacy routing slip flag should be on
+        if routing_slip is None:
+            if not current_app.config.get('ALLOW_LEGACY_ROUTING_SLIPS'):
+                raise BusinessException(Error.RS_DOESNT_EXIST)
+            # legacy routing slip which doesnt exist in the system.No validations
+            return
+
+        # check rs is active
+
+        if routing_slip.status in (
```

Comment:
May be we can check any status other than ACTIVE and LINKED ?

Reply:
done..thats a good idea